src/protocols/disk_io2.rs

```diff
@@ -0,0 +1,52 @@
+//! Disk I/O 2 Protocol
+//!
+//! Extends the Disk I/O protocol interface to enable non-blocking /
+//! asynchronous byte-oriented disk operation.
+
+pub const PROTOCOL_GUID: crate::base::Guid = crate::base::Guid::from_fields(
+    0x151c8eae,
+    0x7f2c,
+    0x472c,
+    0x9e,
+    0x54,
+    &[0x98, 0x28, 0x19, 0x4f, 0x6a, 0x88],
+);
+
+pub const REVISION: u64 = 0x0000000000020000u64;
+
+#[repr(C)]
+#[derive(Debug)]
+pub struct DiskIo2Token {
```

Comment:
I think this should be called `Token`, since we always try to strip the protocol names from all symbols (they are already provided by rust namespacing/modules).